### ISSUE TITLE
[Bug]: Small Scatter Plot Marker Size Results in Circles

### ISSUE THREAD

**Issue by suuuehgi** [matplotlib/matplotlib#25410]
### Bug summary

Using marker sizes `s` smaller than 1 on scatter plots result in hollow circles:
```python
ax.scatter(X, Y, s=.05, c='white', marker='.')
```
![screenshot-2023-03-08-16:39:50](https://user-images.githubusercontent.com/7861774/223760168-117c689f-2ef6-46e0-a3ef-10f1850bc828.png)

### Code for reproduction

```python
from matplotlib import pyplot as plt

S = [4, 3, 2, 1, .5, .1, .01, .001]
N = len(S)

fig, ax = plt.subplots( figsize=(1,.2), layout='constrained')

ax.scatter(range(N), N*[0], s=S)
ax.axis('off')
```

### Actual outcome

![so](https://user-images.githubusercontent.com/7861774/223764737-72bf0c3f-0fad-48bc-acbb-3172c8430a94.jpg)

### Expected outcome

Solid circles

### Additional information

I want to plot streamlines on `pyplot.tricontourf` and for reasons beyond me `s=1` is already way to big (compare above).
I can't easily use the lines of `pyplot.plot` as Comsol just gives me a chunk of (x, y)-pairs.

### Operating system

Arch

### Matplotlib Version

3.5.3

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

Python 3.10.9

### Jupyter version

3.6.1

### Installation

Linux package manager

**Comment by jklymak** [comment]
I think you have left out that you are using svg to render to the inline backend?  

Regardless I think `edgecolors='none'` fixes your problem

**Comment by suuuehgi** [comment]
>I think you have left out that you are using svg to render to the inline backend?

Just in the upper example image. Not in the lower one.

>Regardless I think `edgecolors='none'` fixes your problem.

Nice!! Indeed! Thank you so much!

I leave the issue open for now as it's still inconsistent behavior to me.
You can better assess whether this is a low priority wont-fix or whatever. :)

**Comment by jklymak** [comment]
> Just in the upper example image. Not in the lower one.

How did you make the lower image?  If I use png to render, the image is antialiased, but there is no extra artifact.

> I leave the issue open for now as it's still inconsistent behavior to me.

I'm not quite sure what happens - at some point the edge circle doesn't get any smaller, while the center of the marker does get smaller.  Its possible this is a round off error in our code, or a fundamental limitation of the vector backend.s

**Comment by anntzer** [comment]
I suspect this is just due to the fact that markers are drawn by default with a nonzero edge width - does setting linewidths to 0 fix the problem?

**Comment by tacaswell** [comment]
> fig, ax = plt.subplots( figsize=(1,.2), layout='constrained')

One thing that is going on here is that for...reasons, the `s` input is in pixels squared which for the vector backends is very likely relative to 72dpi.  If your figure is only an inch high we are all viewing these at massive zoom factors.

**Comment by suuuehgi** [comment]
> How did you make the lower image? If I use png to render, the image is antialiased, but there is no extra artifact.

```python
fig.savefig('/tmp/so.jpg', dpi=600)
```

> for...reasons

😁

> If your figure is only an inch high wide and .2in high

This one is just for demonstration purpose. The original figure I cropped the upper image from was about 6 x 2 in.

**Comment by tacaswell** [comment]
TL;DR; This is an artifact of when you draw a path with radius that is smaller than the stroke width of its edge.  The work-around here is to set `lw=0` in the call to scatter to turn the edges off


-----

```python

from matplotlib import pyplot as plt

S = [4, 3, 2, 1, .5, .1, .01, .001]
N = len(S)

fig, ax = plt.subplots( figsize=(1,.5), layout='constrained')

ax.scatter(range(N), N*[0], s=S, edgecolor='red', lw=0)
ax.scatter(range(N), N*[1], s=S, edgecolor='red', lw=1)
ax.scatter(range(N), N*[2], s=S, edgecolor='red', lw=2)
ax.set_ylim(-1, 3)
ax.axis('off')
fig.savefig('/tmp/test.svg')
fig.savefig('/tmp/test.png', dpi=1000)
```

![svg output of the example code showing odd behavior on edges of markers](https://user-images.githubusercontent.com/199813/223852166-99b34f0c-ef2d-4180-8a61-af75488e3149.svg)

Hiding the literal svg in a fold:

<details>

```svg
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
  "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg xmlns:xlink="http://www.w3.org/1999/xlink" width="72pt" height="36pt" viewBox="0 0 72 36" xmlns="http://www.w3.org/2000/svg" version="1.1">
 <metadata>
  <rdf:RDF xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
   <cc:Work>
    <dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"/>
    <dc:date>2023-03-08T16:16:53.578953</dc:date>
    <dc:format>image/svg+xml</dc:format>
    <dc:creator>
     <cc:Agent>
      <dc:title>Matplotlib v3.8.0.dev497+g84e5f0251c.d20230308, https://matplotlib.org/</dc:title>
     </cc:Agent>
    </dc:creator>
   </cc:Work>
  </rdf:RDF>
 </metadata>
 <defs>
  <style type="text/css">*{stroke-linejoin: round; stroke-linecap: butt}</style>
 </defs>
 <g id="figure_1">
  <g id="patch_1">
   <path d="M 0 36 
L 72 36 
L 72 0 
L 0 0 
z
" style="fill: #ffffff"/>
  </g>
  <g id="axes_1">
   <g id="PathCollection_1">
    <path d="M 6.000218 26.49988 
C 6.265421 26.49988 6.519798 26.394514 6.707325 26.206987 
C 6.894852 26.01946 7.000218 25.765083 7.000218 25.49988 
C 7.000218 25.234677 6.894852 24.9803 6.707325 24.792773 
C 6.519798 24.605246 6.265421 24.49988 6.000218 24.49988 
C 5.735015 24.49988 5.480638 24.605246 5.293111 24.792773 
C 5.105584 24.9803 5.000218 25.234677 5.000218 25.49988 
C 5.000218 25.765083 5.105584 26.01946 5.293111 26.206987 
C 5.480638 26.394514 5.735015 26.49988 6.000218 26.49988 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
    <path d="M 14.571584 26.365905 
C 14.801257 26.365905 15.021554 26.274656 15.183957 26.112252 
C 15.34636 25.949849 15.43761 25.729553 15.43761 25.49988 
C 15.43761 25.270207 15.34636 25.049911 15.183957 24.887508 
C 15.021554 24.725104 14.801257 24.633855 14.571584 24.633855 
C 14.341912 24.633855 14.121615 24.725104 13.959212 24.887508 
C 13.796809 25.049911 13.705559 25.270207 13.705559 25.49988 
C 13.705559 25.729553 13.796809 25.949849 13.959212 26.112252 
C 14.121615 26.274656 14.341912 26.365905 14.571584 26.365905 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
    <path d="M 23.142951 26.206987 
C 23.330478 26.206987 23.510349 26.132482 23.642951 25.99988 
C 23.775552 25.867278 23.850057 25.687407 23.850057 25.49988 
C 23.850057 25.312353 23.775552 25.132482 23.642951 24.99988 
C 23.510349 24.867278 23.330478 24.792773 23.142951 24.792773 
C 22.955424 24.792773 22.775552 24.867278 22.642951 24.99988 
C 22.510349 25.132482 22.435844 25.312353 22.435844 25.49988 
C 22.435844 25.687407 22.510349 25.867278 22.642951 25.99988 
C 22.775552 26.132482 22.955424 26.206987 23.142951 26.206987 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
    <path d="M 31.714317 25.99988 
C 31.846918 25.99988 31.974107 25.947197 32.06787 25.853433 
C 32.161634 25.75967 32.214317 25.632482 32.214317 25.49988 
C 32.214317 25.367278 32.161634 25.24009 32.06787 25.146327 
C 31.974107 25.052563 31.846918 24.99988 31.714317 24.99988 
C 31.581715 24.99988 31.454527 25.052563 31.360763 25.146327 
C 31.267 25.24009 31.214317 25.367278 31.214317 25.49988 
C 31.214317 25.632482 31.267 25.75967 31.360763 25.853433 
C 31.454527 25.947197 31.581715 25.99988 31.714317 25.99988 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
    <path d="M 40.285683 25.853433 
C 40.379447 25.853433 40.469382 25.816181 40.535683 25.74988 
C 40.601984 25.683579 40.639237 25.593643 40.639237 25.49988 
C 40.639237 25.406117 40.601984 25.316181 40.535683 25.24988 
C 40.469382 25.183579 40.379447 25.146327 40.285683 25.146327 
C 40.19192 25.146327 40.101984 25.183579 40.035683 25.24988 
C 39.969382 25.316181 39.93213 25.406117 39.93213 25.49988 
C 39.93213 25.593643 39.969382 25.683579 40.035683 25.74988 
C 40.101984 25.816181 40.19192 25.853433 40.285683 25.853433 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
    <path d="M 48.857049 25.657994 
C 48.898982 25.657994 48.939202 25.641334 48.968853 25.611683 
C 48.998503 25.582033 49.015163 25.541812 49.015163 25.49988 
C 49.015163 25.457948 48.998503 25.417727 48.968853 25.388077 
C 48.939202 25.358426 48.898982 25.341766 48.857049 25.341766 
C 48.815117 25.341766 48.774897 25.358426 48.745246 25.388077 
C 48.715595 25.417727 48.698935 25.457948 48.698935 25.49988 
C 48.698935 25.541812 48.715595 25.582033 48.745246 25.611683 
C 48.774897 25.641334 48.815117 25.657994 48.857049 25.657994 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
    <path d="M 57.428416 25.54988 
C 57.441676 25.54988 57.454395 25.544612 57.463771 25.535235 
C 57.473147 25.525859 57.478416 25.51314 57.478416 25.49988 
C 57.478416 25.48662 57.473147 25.473901 57.463771 25.464525 
C 57.454395 25.455148 57.441676 25.44988 57.428416 25.44988 
C 57.415155 25.44988 57.402437 25.455148 57.39306 25.464525 
C 57.383684 25.473901 57.378416 25.48662 57.378416 25.49988 
C 57.378416 25.51314 57.383684 25.525859 57.39306 25.535235 
C 57.402437 25.544612 57.415155 25.54988 57.428416 25.54988 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
    <path d="M 65.999782 25.515691 
C 66.003975 25.515691 66.007997 25.514025 66.010962 25.51106 
C 66.013927 25.508095 66.015593 25.504073 66.015593 25.49988 
C 66.015593 25.495687 66.013927 25.491665 66.010962 25.4887 
C 66.007997 25.485735 66.003975 25.484069 65.999782 25.484069 
C 65.995589 25.484069 65.991567 25.485735 65.988601 25.4887 
C 65.985636 25.491665 65.98397 25.495687 65.98397 25.49988 
C 65.98397 25.504073 65.985636 25.508095 65.988601 25.51106 
C 65.991567 25.514025 65.995589 25.515691 65.999782 25.515691 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
   </g>
   <g id="PathCollection_2">
    <path d="M 6.000218 19 
C 6.265421 19 6.519798 18.894634 6.707325 18.707107 
C 6.894852 18.51958 7.000218 18.265203 7.000218 18 
C 7.000218 17.734797 6.894852 17.48042 6.707325 17.292893 
C 6.519798 17.105366 6.265421 17 6.000218 17 
C 5.735015 17 5.480638 17.105366 5.293111 17.292893 
C 5.105584 17.48042 5.000218 17.734797 5.000218 18 
C 5.000218 18.265203 5.105584 18.51958 5.293111 18.707107 
C 5.480638 18.894634 5.735015 19 6.000218 19 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
    <path d="M 14.571584 18.866025 
C 14.801257 18.866025 15.021554 18.774776 15.183957 18.612372 
C 15.34636 18.449969 15.43761 18.229673 15.43761 18 
C 15.43761 17.770327 15.34636 17.550031 15.183957 17.387628 
C 15.021554 17.225224 14.801257 17.133975 14.571584 17.133975 
C 14.341912 17.133975 14.121615 17.225224 13.959212 17.387628 
C 13.796809 17.550031 13.705559 17.770327 13.705559 18 
C 13.705559 18.229673 13.796809 18.449969 13.959212 18.612372 
C 14.121615 18.774776 14.341912 18.866025 14.571584 18.866025 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
    <path d="M 23.142951 18.707107 
C 23.330478 18.707107 23.510349 18.632602 23.642951 18.5 
C 23.775552 18.367398 23.850057 18.187527 23.850057 18 
C 23.850057 17.812473 23.775552 17.632602 23.642951 17.5 
C 23.510349 17.367398 23.330478 17.292893 23.142951 17.292893 
C 22.955424 17.292893 22.775552 17.367398 22.642951 17.5 
C 22.510349 17.632602 22.435844 17.812473 22.435844 18 
C 22.435844 18.187527 22.510349 18.367398 22.642951 18.5 
C 22.775552 18.632602 22.955424 18.707107 23.142951 18.707107 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
    <path d="M 31.714317 18.5 
C 31.846918 18.5 31.974107 18.447317 32.06787 18.353553 
C 32.161634 18.25979 32.214317 18.132602 32.214317 18 
C 32.214317 17.867398 32.161634 17.74021 32.06787 17.646447 
C 31.974107 17.552683 31.846918 17.5 31.714317 17.5 
C 31.581715 17.5 31.454527 17.552683 31.360763 17.646447 
C 31.267 17.74021 31.214317 17.867398 31.214317 18 
C 31.214317 18.132602 31.267 18.25979 31.360763 18.353553 
C 31.454527 18.447317 31.581715 18.5 31.714317 18.5 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
    <path d="M 40.285683 18.353553 
C 40.379447 18.353553 40.469382 18.316301 40.535683 18.25 
C 40.601984 18.183699 40.639237 18.093763 40.639237 18 
C 40.639237 17.906237 40.601984 17.816301 40.535683 17.75 
C 40.469382 17.683699 40.379447 17.646447 40.285683 17.646447 
C 40.19192 17.646447 40.101984 17.683699 40.035683 17.75 
C 39.969382 17.816301 39.93213 17.906237 39.93213 18 
C 39.93213 18.093763 39.969382 18.183699 40.035683 18.25 
C 40.101984 18.316301 40.19192 18.353553 40.285683 18.353553 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
    <path d="M 48.857049 18.158114 
C 48.898982 18.158114 48.939202 18.141454 48.968853 18.111803 
C 48.998503 18.082153 49.015163 18.041932 49.015163 18 
C 49.015163 17.958068 48.998503 17.917847 48.968853 17.888197 
C 48.939202 17.858546 48.898982 17.841886 48.857049 17.841886 
C 48.815117 17.841886 48.774897 17.858546 48.745246 17.888197 
C 48.715595 17.917847 48.698935 17.958068 48.698935 18 
C 48.698935 18.041932 48.715595 18.082153 48.745246 18.111803 
C 48.774897 18.141454 48.815117 18.158114 48.857049 18.158114 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
    <path d="M 57.428416 18.05 
C 57.441676 18.05 57.454395 18.044732 57.463771 18.035355 
C 57.473147 18.025979 57.478416 18.01326 57.478416 18 
C 57.478416 17.98674 57.473147 17.974021 57.463771 17.964645 
C 57.454395 17.955268 57.441676 17.95 57.428416 17.95 
C 57.415155 17.95 57.402437 17.955268 57.39306 17.964645 
C 57.383684 17.974021 57.378416 17.98674 57.378416 18 
C 57.378416 18.01326 57.383684 18.025979 57.39306 18.035355 
C 57.402437 18.044732 57.415155 18.05 57.428416 18.05 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
    <path d="M 65.999782 18.015811 
C 66.003975 18.015811 66.007997 18.014145 66.010962 18.01118 
C 66.013927 18.008215 66.015593 18.004193 66.015593 18 
C 66.015593 17.995807 66.013927 17.991785 66.010962 17.98882 
C 66.007997 17.985855 66.003975 17.984189 65.999782 17.984189 
C 65.995589 17.984189 65.991567 17.985855 65.988601 17.98882 
C 65.985636 17.991785 65.98397 17.995807 65.98397 18 
C 65.98397 18.004193 65.985636 18.008215 65.988601 18.01118 
C 65.991567 18.014145 65.995589 18.015811 65.999782 18.015811 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
   </g>
   <g id="PathCollection_3">
    <path d="M 6.000218 11.50012 
C 6.265421 11.50012 6.519798 11.394754 6.707325 11.207227 
C 6.894852 11.0197 7.000218 10.765323 7.000218 10.50012 
C 7.000218 10.234917 6.894852 9.98054 6.707325 9.793013 
C 6.519798 9.605486 6.265421 9.50012 6.000218 9.50012 
C 5.735015 9.50012 5.480638 9.605486 5.293111 9.793013 
C 5.105584 9.98054 5.000218 10.234917 5.000218 10.50012 
C 5.000218 10.765323 5.105584 11.0197 5.293111 11.207227 
C 5.480638 11.394754 5.735015 11.50012 6.000218 11.50012 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
    <path d="M 14.571584 11.366145 
C 14.801257 11.366145 15.021554 11.274896 15.183957 11.112492 
C 15.34636 10.950089 15.43761 10.729793 15.43761 10.50012 
C 15.43761 10.270447 15.34636 10.050151 15.183957 9.887748 
C 15.021554 9.725344 14.801257 9.634095 14.571584 9.634095 
C 14.341912 9.634095 14.121615 9.725344 13.959212 9.887748 
C 13.796809 10.050151 13.705559 10.270447 13.705559 10.50012 
C 13.705559 10.729793 13.796809 10.950089 13.959212 11.112492 
C 14.121615 11.274896 14.341912 11.366145 14.571584 11.366145 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
    <path d="M 23.142951 11.207227 
C 23.330478 11.207227 23.510349 11.132722 23.642951 11.00012 
C 23.775552 10.867518 23.850057 10.687647 23.850057 10.50012 
C 23.850057 10.312593 23.775552 10.132722 23.642951 10.00012 
C 23.510349 9.867518 23.330478 9.793013 23.142951 9.793013 
C 22.955424 9.793013 22.775552 9.867518 22.642951 10.00012 
C 22.510349 10.132722 22.435844 10.312593 22.435844 10.50012 
C 22.435844 10.687647 22.510349 10.867518 22.642951 11.00012 
C 22.775552 11.132722 22.955424 11.207227 23.142951 11.207227 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
    <path d="M 31.714317 11.00012 
C 31.846918 11.00012 31.974107 10.947437 32.06787 10.853673 
C 32.161634 10.75991 32.214317 10.632722 32.214317 10.50012 
C 32.214317 10.367518 32.161634 10.24033 32.06787 10.146567 
C 31.974107 10.052803 31.846918 10.00012 31.714317 10.00012 
C 31.581715 10.00012 31.454527 10.052803 31.360763 10.146567 
C 31.267 10.24033 31.214317 10.367518 31.214317 10.50012 
C 31.214317 10.632722 31.267 10.75991 31.360763 10.853673 
C 31.454527 10.947437 31.581715 11.00012 31.714317 11.00012 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
    <path d="M 40.285683 10.853673 
C 40.379447 10.853673 40.469382 10.816421 40.535683 10.75012 
C 40.601984 10.683819 40.639237 10.593883 40.639237 10.50012 
C 40.639237 10.406357 40.601984 10.316421 40.535683 10.25012 
C 40.469382 10.183819 40.379447 10.146567 40.285683 10.146567 
C 40.19192 10.146567 40.101984 10.183819 40.035683 10.25012 
C 39.969382 10.316421 39.93213 10.406357 39.93213 10.50012 
C 39.93213 10.593883 39.969382 10.683819 40.035683 10.75012 
C 40.101984 10.816421 40.19192 10.853673 40.285683 10.853673 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
    <path d="M 48.857049 10.658234 
C 48.898982 10.658234 48.939202 10.641574 48.968853 10.611923 
C 48.998503 10.582273 49.015163 10.542052 49.015163 10.50012 
C 49.015163 10.458188 48.998503 10.417967 48.968853 10.388317 
C 48.939202 10.358666 48.898982 10.342006 48.857049 10.342006 
C 48.815117 10.342006 48.774897 10.358666 48.745246 10.388317 
C 48.715595 10.417967 48.698935 10.458188 48.698935 10.50012 
C 48.698935 10.542052 48.715595 10.582273 48.745246 10.611923 
C 48.774897 10.641574 48.815117 10.658234 48.857049 10.658234 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
    <path d="M 57.428416 10.55012 
C 57.441676 10.55012 57.454395 10.544852 57.463771 10.535475 
C 57.473147 10.526099 57.478416 10.51338 57.478416 10.50012 
C 57.478416 10.48686 57.473147 10.474141 57.463771 10.464765 
C 57.454395 10.455388 57.441676 10.45012 57.428416 10.45012 
C 57.415155 10.45012 57.402437 10.455388 57.39306 10.464765 
C 57.383684 10.474141 57.378416 10.48686 57.378416 10.50012 
C 57.378416 10.51338 57.383684 10.526099 57.39306 10.535475 
C 57.402437 10.544852 57.415155 10.55012 57.428416 10.55012 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
    <path d="M 65.999782 10.515931 
C 66.003975 10.515931 66.007997 10.514265 66.010962 10.5113 
C 66.013927 10.508335 66.015593 10.504313 66.015593 10.50012 
C 66.015593 10.495927 66.013927 10.491905 66.010962 10.48894 
C 66.007997 10.485975 66.003975 10.484309 65.999782 10.484309 
C 65.995589 10.484309 65.991567 10.485975 65.988601 10.48894 
C 65.985636 10.491905 65.98397 10.495927 65.98397 10.50012 
C 65.98397 10.504313 65.985636 10.508335 65.988601 10.5113 
C 65.991567 10.514265 65.995589 10.515931 65.999782 10.515931 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
   </g>
  </g>
 </g>
 <defs>
  <clipPath id="p76f3feaf1a">
   <rect x="3.00024" y="3.00024" width="65.99952" height="29.99952"/>
  </clipPath>
 </defs>
</svg>

```

</details>

Pulling out representative snippets for a marker in each row (I grabbed the smallest one, but :

(lw 2)

```svg
    <path d="M 65.999782 10.515931 
C 66.003975 10.515931 66.007997 10.514265 66.010962 10.5113 
C 66.013927 10.508335 66.015593 10.504313 66.015593 10.50012 
C 66.015593 10.495927 66.013927 10.491905 66.010962 10.48894 
C 66.007997 10.485975 66.003975 10.484309 65.999782 10.484309 
C 65.995589 10.484309 65.991567 10.485975 65.988601 10.48894 
C 65.985636 10.491905 65.98397 10.495927 65.98397 10.50012 
C 65.98397 10.504313 65.985636 10.508335 65.988601 10.5113 
C 65.991567 10.514265 65.995589 10.515931 65.999782 10.515931 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #2ca02c; stroke: #ff0000; stroke-width: 2"/>
```

(lw 1)

```svg
    <path d="M 65.999782 18.015811 
C 66.003975 18.015811 66.007997 18.014145 66.010962 18.01118 
C 66.013927 18.008215 66.015593 18.004193 66.015593 18 
C 66.015593 17.995807 66.013927 17.991785 66.010962 17.98882 
C 66.007997 17.985855 66.003975 17.984189 65.999782 17.984189 
C 65.995589 17.984189 65.991567 17.985855 65.988601 17.98882 
C 65.985636 17.991785 65.98397 17.995807 65.98397 18 
C 65.98397 18.004193 65.985636 18.008215 65.988601 18.01118 
C 65.991567 18.014145 65.995589 18.015811 65.999782 18.015811 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #ff7f0e; stroke: #ff0000"/>
```

(no edge)

```svg
    <path d="M 65.999782 25.515691 
C 66.003975 25.515691 66.007997 25.514025 66.010962 25.51106 
C 66.013927 25.508095 66.015593 25.504073 66.015593 25.49988 
C 66.015593 25.495687 66.013927 25.491665 66.010962 25.4887 
C 66.007997 25.485735 66.003975 25.484069 65.999782 25.484069 
C 65.995589 25.484069 65.991567 25.485735 65.988601 25.4887 
C 65.985636 25.491665 65.98397 25.495687 65.98397 25.49988 
C 65.98397 25.504073 65.985636 25.508095 65.988601 25.51106 
C 65.991567 25.514025 65.995589 25.515691 65.999782 25.515691 
z
" clip-path="url(#p76f3feaf1a)" style="fill: #1f77b4"/>
```

The critical differences between the three are in the `style` attribute (and with in a row in `d` attribute) which sets the color of the fill, the color of the stroke (if any), and the width of the stroke.  So, from an svg point of view these things are all ~equivalent and nothing fishy is happening at the specification level.

Looking instead at the png (rendered at 1k dpi) 

![high res png of test image to fake zooming in on svg](https://user-images.githubusercontent.com/199813/223855590-1b05ce10-b8cb-4dd2-9c7d-64cc2b3be942.png) we see the same effects where in the case with 0 width edges we get the expected behavior, but in the case of a finite width edge we see some weirdness.  Looking at the middle row (lw = 1) we see the as the marker gets smaller the edge goes from being and edge to fully overlapping the fill region.  This makes sense as the path on a stroke goes down the center so for a small enough circle the radius is equal to half the stroke width and we get a "solid" circle of the edge color.   However when we go smaller I do not have a clear understanding of _why_ it is doing this, but it appears that as the edge line starts to go "out the other side" we get an effect of the edge getting _narrower_ visually (even though it is fixed in the svg and I expect it to be fixed in Agg).   When the circle is significantly smaller than edge width we also start to see some star like artifacts which I also do not strictly understand.

I think the alternate behavior (and maybe what I would have expected?) is for the markers to become a fixed size (at effectively the edge width in diameter), but I strongly suspect that would have bad artifacts on very sharp corners (turning them into dots rather than points).

**Comment by tacaswell** [comment]
https://github.com/matplotlib/matplotlib/issues/17424 is plausibly related (as the svg circle primitive _may_ behave better).

**Comment by anntzer** [comment]
http://tavmjong.free.fr/blog/?p=1257 and https://www.w3.org/TR/SVG2/painting.html#StrokeShape may be related too.

**Comment by tacaswell** [comment]
Indeed, based on the first link, I suspect offset paths + fill between is what explains the doughnut effect.

**Comment by idanpa** [comment]
For anyone that got here due to the same problem in `ax.plot`,  adding `markeredgewidth=0.0` solves the problem.
